### PR TITLE
Split read/write record buffer to support buffered reads

### PR DESCRIPTION
**Pull request by rmja**

### CODE REVIEW

File: src/asynch.rs [hunk 1]
```diff
@@ -40,14 +44,29 @@ where
...
+    /// The write record buffer can be smaller than the read buffer. During write [`TLS_RECORD_OVERHEAD`] over overhead
+    /// is added per record, so the buffer must at least be this large. Large writes are split into multiple records if
+    /// depending on the size of the write buffer.
+    /// The largest of the two buffers will be used to encode the TLS handshake record, hence either of the
+    /// buffers must at least be large enough to encode a handshake.
+    pub fn new(
+        delegate: Socket,
+        record_read_buf: &'a mut [u8],
+        record_write_buf: &'a mut [u8],
+    ) -> Self {
```
Comment (lulf) [inline src/asynch.rs:60]: We should add an assertion here that record_write_buf.len() >= TLS_RECORD_OVERHEAD.

Comment (lulf) [reply src/asynch.rs:60]: Likewise, at least one of them should fit the minimal size of the handshake, but it's a bit more complex to determine so can be left out for now.

File: src/asynch.rs [hunk 2]
```diff
@@ -125,61 +149,88 @@ where
...
+                    assert!(offset + slice.len() <= buf_len);
+
+                    self.decrypted_offset = offset;
+                    self.decrypted_len = slice.len();
+                    self.decrypted_consumed = 0;
+                    return Ok(data.data);
+                }
+                ServerRecord::Alert(alert) => {
+                    if let AlertDescription::CloseNotify = alert.description {
+                        Err(TlsError::ConnectionClosed)
```
Comment (lulf) [inline src/asynch.rs:216]: Re: our discussion earlier, this should probably set self.opened = false;

File: src/asynch.rs (hunk 1 above)
Comment (rmja) [reply src/asynch.rs:60]: I added the assertion

File: src/asynch.rs (hunk 2 above)
Comment (rmja) [reply src/asynch.rs:221]: Done

**Review by lulf — APPROVED** [review]
LGTM, thanks for fixing this 🚀 

Closes #44

**Pull request merged**